### PR TITLE
SAK-33971 - Assignments List: Filter assignments by group

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -11815,6 +11844,20 @@ protected int sizeResources(SessionState state) {
...
                 }
 
+                //Filter assignments by group
+                if(StringUtils.isNotEmpty(selectedGroup) && !"all".equals(selectedGroup)){
+                    List<Assignment> filteredAssignments = new ArrayList<Assignment>();
+                    for(Assignment assignment : (List<Assignment>) returnResources){
+                        Collection<String> groups = assignment.getGroups();
+                        for(String group : groups){
+                            if(group.endsWith(selectedGroup)){
+                                filteredAssignments.add(assignment);
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: lambda might be nice here:
List<Assignment> filteredAssignments = assignments.stream().filter(a -> a.getGroups().stream().anyMatch(g -> g.endsWith("something"))).collect(Collectors.toList());

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -873,6 +877,10 @@
      * Site property for forcing anonymous grading in a site
      */
     private static final String SAK_PROP_FORCE_ANON_GRADING = "assignment.anon.grading.forced";
+    /** 
+     * Sakai.property for enable/disable the group filter
+     */
+    private static final String SAK_PROP_ENABLE_GROUP_FILTER = "assignment.group.filter.enabled";
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: This property should be in AssignmentServiceConstants.

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 3]
```diff
@@ -11815,6 +11844,20 @@ protected int sizeResources(SessionState state) {
                     returnResources.addAll(assignmentService.getAssignmentsForContext((String) state.getAttribute(STATE_CONTEXT_STRING)));
                 }
 
+                //Filter assignments by group
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: eventually sizeResources will be replaced with a more efficient method in the service!

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 2 above)
Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Good to know, thanks!

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 3 above)
Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Yeah, makes sense

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 1 above)
Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Looks good, thank you!

Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Definitely, I need a fast course of Lambda Expressions, I like them, they are really really powerful.

**Pull request merged**